src/rust/ide/src/controller/visualization.rs

```diff
@@ -0,0 +1,262 @@
+use graph_editor::component::visualization::JsSourceClass;
+
+
+
+// =============
+// === Error ===
+// =============
+
+/// Enumeration of errors used in `Visualization Controller`.
+#[derive(Debug,Fail)]
+#[allow(missing_docs)]
+pub enum VisualizationError {
+    #[fail(display = "Visualization \"{}\" not found.", identifier)]
+    NotFound {
+        identifier : VisualizationIdentifier
+    },
+    #[fail(display = "JavaScript visualization \"{}\" failed to compile.", identifier)]
+    CompileError {
+        identifier : VisualizationIdentifier
+    }
+}
+
+
+
+// ===============================
+// === VisualizationIdentifier ===
+// ===============================
+
+/// This enum is used to identify a visualiser both in the project folder or natively embedded in
+/// IDE.
+#[derive(Clone,Debug,Display,Eq,PartialEq)]
+#[allow(missing_docs)]
+pub enum VisualizationIdentifier {
+    Embedded(String),
+    File(language_server::Path)
+}
+
+
+
+// ==============================
+// === EmbeddedVisualizations ===
+// ==============================
+
+#[allow(missing_docs)]
+pub type EmbeddedVisualizationName   = String;
+
+/// Embedded visualizations mapped from name to source code.
+#[derive(Shrinkwrap,Debug,Clone,Default)]
+#[shrinkwrap(mutable)]
+pub struct EmbeddedVisualizations {
+    #[allow(missing_docs)]
+    pub map : HashMap<EmbeddedVisualizationName,Rc<class::Handle>>
+}
+
+
+
+
+// ==============
+// === Handle ===
+// ==============
+
+const VISUALISATION_FOLDER : &str = "visualization";
+
+/// Visualization Controller's state.
+#[derive(Debug,Clone)]
+pub struct Handle {
+    language_server_rpc     : Rc<language_server::Connection>,
+    embedded_visualizations : EmbeddedVisualizations,
+    graph_editor            : RefCell<Option<GraphEditor>>
+}
+
+impl Handle {
+    /// Creates a new visualization controller.
+    pub fn new
+    ( language_server_rpc     : Rc<language_server::Connection>
+    , embedded_visualizations : EmbeddedVisualizations) -> Self {
+        let graph_editor = RefCell::new(None);
+        Self {language_server_rpc,embedded_visualizations,graph_editor}
+    }
+
+    /// Sets the GraphEditor to report about visualizations availability.
+    pub async fn set_graph_editor(&self, graph_editor:Option<GraphEditor>) -> FallibleResult<()> {
+        let identifiers = self.list_visualizations().await;
+        let identifiers = identifiers.unwrap_or_default();
+        for identifier in identifiers {
+            let visualization = self.load_visualization(&identifier).await;
+            visualization.map(|visualization| {
+                if let Some(graph_editor) = graph_editor.as_ref() {
+                    let class_handle = &Some(visualization);
+                    graph_editor.frp.register_visualization_class.emit_event(class_handle);
+                }
+            })?;
+        }
+        *self.graph_editor.borrow_mut() = graph_editor;
+        Ok(())
+    }
```

Comment:
This is code which should be moved to `view::node_editor` module (or some submodule of it).